components/camel-jt400/src/main/java/org/apache/camel/component/jt400/Jt400MsgQueueProducer.java

```diff
@@ -50,7 +50,12 @@ public void process(Exchange exchange) throws Exception {
 
     private void process(MessageQueue queue, Exchange exchange) throws Exception {
         String msgText = exchange.getIn().getBody(String.class);
-        queue.sendInformational(msgText);
+        byte[] messageKey = (byte[]) exchange.getIn().getHeader(Jt400Constants.MESSAGE_REPLYTO_KEY);
```

Comment:
You can actually do something like this here instead of the explicit casting
```suggestion
        byte[] messageKey =  exchange.getIn().getHeader(Jt400Constants.MESSAGE_REPLYTO_KEY, byte[].class);
```